src/components/BookingModal/container.js

```diff
@@ -169,27 +187,54 @@ const Container = Wrapped =>
         });
     };
 
+    getApprovedDays = () => {
+      return getTotalDaysInEventArrayWithStatus(
+        this.props.allEvents,
+        HolidayStatus.APPROVED
+      );
+    };
+
+    getPendingDays = () => {
+      return getTotalDaysInEventArrayWithStatus(
+        this.props.allEvents,
+        HolidayStatus.PENDING
+      );
+    };
+
+    onFormUpdate = formData => {
+      this.setState({ formData });
+    };
+
     render() {
       const { toggleRejectionMessageView, loading } = this.state;
+      const {
+        booking: { start },
+      } = this.props;
       return (
         this.props.employeeId && (
           <Wrapped
-            legacyHolidayMessagelist={this.legacyHolidayMessagelist}
-            toggleRejectionMessageView={toggleRejectionMessageView}
-            rejectionResponseText={this.state.rejectionResponseText}
-            assignRejectionResponseText={this.assignRejectionResponseText}
+            workingFromHomeBooking={checkSameDate(start)}
+            formData={this.state.formData}
+            userDetails={this.props.userDetails}
+            pendingDays={this.getPendingDays()}
+            approvedDays={this.getApprovedDays()}
             booking={this.props.booking}
-            toggleLegacyHolidayMessageView={this.toggleLegacyHolidayMessageView}
             employeeId={this.props.employeeId}
             bookingModalOpen={this.props.bookingModalOpen}
             closeBookingModal={this.closeBookingModal}
-            updateTakenEvents={this.props.updateTakenEvents}
-            isEventBeingUpdated={this.props.isEventBeingUpdated}
-            bookingDuration={this.props.bookingDuration}
             createEvent={this.createEvent}
             updateEvent={this.updateEvent}
             cancelEvent={this.cancelEvent}
             loading={loading}
+            onFormUpdate={this.onFormUpdate}
+            updateTakenEvents={this.props.updateTakenEvents}
+            bookingDuration={this.props.bookingDuration}
+            isEventBeingUpdated={this.props.isEventBeingUpdated}
+            legacyHolidayMessagelist={this.legacyHolidayMessagelist}
+            toggleRejectionMessageView={toggleRejectionMessageView}
+            rejectionResponseText={this.state.rejectionResponseText}
+            assignRejectionResponseText={this.assignRejectionResponseText}
+            toggleLegacyHolidayMessageView={this.toggleLegacyHolidayMessageView}
```

Comment:
Hot dog that's a lot of props.

Reply:
Part of the problem we're trying to solve, but we paused to get everyone on the same page again. I'm not sure all this logic should be in this file either. Will be one to look at next week.

Reply:
I probably need to have the reasoning behind this architecture explained to me. The whole thing seems overly complex (Not a problem of Michaela's, more of an overarching thing)

Reply:
Just to clarify this isn't Michaela's code - it was existing code that has just changed lines so Github is picking it up as an addition.